addons/web_editor/static/src/js/wysiwyg/wysiwyg_iframe.js

```diff
@@ -207,7 +207,7 @@ patch(Wysiwyg.prototype, {
      * @override
      */
     _onScroll(ev) {
-        if (this.options.inIframe) {
+        if (this.options.inIframe || this.options.isStudio) {
```

Comment:
Why create a new option for that? If studio needs the `inIframe` behavior, why not set `inIframe` to `true` in studio?

Reply:
When `inIframe` is true, in `wysiwyg_iframe` the `startEdition` method calls `this._loadIframe`, which in turn calls the `onIframeUpdated` method. However, since this method is only available in `mass_mailing`, a traceback occurs.

Reply:
The code that calls `onIframeUpdated` should test that it exists before calling it. Since it is in `this.options`, it does not make any sense to call it without checking that it was defined, since it is optional.

Reply:
The studio report is aleady within an `iframe`, so when `startEdition` is trigger it will make that iframe editable. However, if we set `inIframe` to true, it triggers the `loadIframe` method, which will create a new iframe inside document and then when startEdition is called, it makes that newly created iframe editable instead. As a result, the studio report will not become editable because its original iframe is not set to editable. To address this, if `inIframe` is true, we may need to move the studio iframe or its contents into the newly created iframe.

Reply:
Maybe then the condition for the `_onScroll` should be to check if we are in an iframe rather than checking for the `inIframe` option? Since we can be in an iframe even though `inIframe` is set to `false` (since in practice that option behaves more like if it was named `useIframe` instead of `inIframe`.)

Reply:
Then instead of checking `inIframe` in `_onScroll`, how about using a condition like this: `if (this.$iframe && this.$iframe.length)`?

Reply:
But `this.$iframe` will be `undefined` when `this.options.inIframe` is falsy, which is the case in studio if I understood correctly.

Reply:
That is why I am passing the `iframe` value from the web_studio. Also, we need the `this.$iframe` value inside the if statement; otherwise, both the if and else if conditions will fail.

Reply:
Then maybe we can do something like:

- if `iframe` is given, then `inIframe` is forced to `true`
- if `iframe` has been given, then `inIframe` should use that one and not create a new one

Then checking `inIframe` in `_onScroll` should be enough as long as the system knows to use the already existing iframe.

Reply:
Done!